crates/sui/src/unit_tests/cli_tests.rs

```diff
@@ -852,6 +854,145 @@ async fn test_package_publish_nonexistent_dependency() -> Result<(), anyhow::Err
     Ok(())
 }
 
+#[sim_test]
+async fn test_package_upgrade_command() -> Result<(), anyhow::Error> {
```

Comment:
You may want to add a line in this test that registers our `SuiPackageHooks` which recognizes `published-at`:

```rust
move_package::package_hooks::register_package_hooks(Box::new(SuiPackageHooks {}));
```

This is because we register the hook higher up the code path (in `sui_commands`) for normal execution of the `sui` binary, and this test entry point is a bit further along, in `client_commands`.

Without this line, things will currently still work, but a warning gets dumped to `stderr` about an unrecognized `published-at` field (and gets suppressed in tests).


This is a general annoyance that we/I don't have a concrete solution for dealing with elegantly in tests yet (something like a wrapper would probably work, [more context here](https://github.com/MystenLabs/sui/pull/8902/files#r1126869107))